legacy/firmware/signing.c

```diff
@@ -1487,6 +1553,14 @@ static bool signing_validate_input(const TxInputType *txinput) {
     return false;
   }
 
+  if (!coin->decred && txinput->has_decred_tree) {
```

Comment:
should this be under `#if !BITCOIN_ONLY`, just for neatness' sake?

Reply:
Could be, but what should BTC-only firmware do then in `decred_tree` is specified? Should it ignore or return error?

Reply:
ignore bc from the btc-only point of view the field doesn't exist and the protobuf entry is just the kind of noise protobuf is explicitly allowed to have

Reply:
`core` also doesn't take BTC-only into account in this check.

Reply:
and it probably also should. but ok, let's keep it the same on both sides